tests/helper.ts

```diff
@@ -0,0 +1,312 @@
+import { randomUUID, createHmac } from 'crypto'
+import { stringify } from 'querystring'
+import { faker } from '@faker-js/faker'
+import { IPN_TYPES_ENUM, IPN_DEFAULTS, BASE_IPN_KEYS } from '../src/constants'
+import { ApiFields } from '../src/ipn/api'
+import { AdvancedButtonFields } from '../src/ipn/button'
+import { CartFields } from '../src/ipn/cart'
+import { DepositFields, DepositPaymentFields } from '../src/ipn/deposit'
+import { DonationButtonFields } from '../src/ipn/donation'
+import { SimpleButtonFields } from '../src/ipn/simple'
+import { WithdrawalFields } from '../src/ipn/withdrawal'
+import {
+  BaseIPNFields,
+  BuyerInformationFields,
+  ShippingInformationFields,
+  IPN_TYPES,
+  PaymentStatusFields,
+  CoinpaymentsIPN
+} from '../src/types'
+
+export const createBaseIPNPartial = (): Omit<BaseIPNFields, typeof BASE_IPN_KEYS.TYPE> => ({
+  [BASE_IPN_KEYS.MODE]: IPN_DEFAULTS[BASE_IPN_KEYS.MODE],
```

Comment:
having one big generic helper is an antipattern, better split into cohesive units

Reply:
It is what it is, it works and its done, i regret doing it.

Reply:
ummm, ok